src/Simulation.cpp

```diff
@@ -682,11 +683,15 @@ void Simulation::readXML(XMLfileUnits& xmlconfig) {
 		bool ignoreCheckpointTime = false;
 		if(xmlconfig.getNodeValue("ignoreCheckpointTime", ignoreCheckpointTime)) {
 			if(ignoreCheckpointTime)
-				_simulationTime = 0;
+				setSimulationTime(0.0);
 		}
 	}
 
 	xmlconfig.changecurrentnode(oldpath);
+
+	// _simulationTime might be updated by readers -> also update _initSimulation
+	_simstep = _initSimulation = (unsigned long) round(_simulationTime / _integrator->getTimestepLength() );
```

Comment:
```suggestion
	_simstep = _initSimulation = static_cast<unsigned long>(std::round(_simulationTime / _integrator->getTimestepLength()));
```
Is `std::round` really what we want here? Not `std::floor`?

Reply:
I think that `round` is better, since `floor` in combination with numerical errors or time-rounded-in-checkpoint-file errors can lead to wrong/unintended results. Why would you prefer `floor`?

Reply:
Say `_simulationTime = 8` and `_integrator->getTimestepLength() == 3`, then the value before the rounding is `2.6`. Does it make sense to say we start with simstep 2 or 3 in this case?

Wouldn't it make more sense to start with the next simstep after the last from the previous simulation? Otherwise, we might "redo" sim step 2 or "skip" it.

When is this code even relevant? I thought loading the checkpoint directly updates these variables?

Reply:
Yes, the goal is to continue with the step after the last one from the previous simulation. Therefore, the last time is stored in the checkpoint headers (e.g. `<time>1.094204099996757e+03</time>`). Let say a typical time step width is `0.003647347`. This will then result in a last "time step" of time/width = `299999.999999`. If you now floor this value, the initial time step will be wrong.

There is already the issue with "redoing" the last simstep of a checkpoint. But I think this is not a big deal, since there is also a simulation step `0`.

Reply:
But wouldn't then, by this argument, `std::ceil` be more along what we want?

Reply:
Not sure. Maybe the calculation result is sometimes `300000.000001`. In all cases, `round` would work best, I think.

Reply:
Ok then let's go for that and reevaluate should issues arise. Just make sure to leave a comment on the code why you chose `round`.